### ISSUE TITLE
Multi $expand returns 500 error on V2+ APIs for .NET Core OData w/ Swagger

### ISSUE THREAD

**Issue by Clebeuf** [microsoft/aspnet-api-versioning#631]
I used the SwaggerODataSample (https://github.com/microsoft/aspnet-api-versioning/tree/master/samples/aspnetcore/SwaggerODataSample) as a template to create a Versioned OData API with .NET Core and Entity Framework DB. 

Our API has multiple $expands configured for an endpoint. This works perfectly for V1, but doesn't work for V2+. 

API configuration:

```
namespace Sample.API.Configuration
{
    static class ApiVersions
    {
        internal static readonly ApiVersion V1 = new ApiVersion( 1, 0 );
        internal static readonly ApiVersion V2 = new ApiVersion( 2, 0 );
    }
}
```

Simplified Controller Code:

```
namespace Sample.API
{
    [ApiVersion("1.0")]
    [ApiVersion("2.0")]
    [ODataRoutePrefix("X")]
    public class XController : ODataController 
    {
        [ODataRoute]
        [Produces("application/json")]
        [ProducesResponseType(typeof(List<X>), Status200OK)]
        [EnableQuery(MaxTop = 500, AllowedQueryOptions = Filter | Select | OrderBy | Top | Skip | Expand | Count)]
        public IQueryable<X> Get()
        {
            return <EntityFrameworkContext>.X;
        }

        [ODataRoute("('{key}')")]
        [Produces("application/json")]
        [ProducesResponseType(typeof(X), Status200OK)]
        [ProducesResponseType(Status404NotFound)]
        [EnableQuery(MaxTop = 500, AllowedQueryOptions = Filter | Select | Expand)]
        public SingleResult<X> Get(string key)
        {
            return SingleResult.Create(<EntityFrameworkContext>.X.Where(g => g.key == key));
        }
    }
}
```

Tables Y & Z both have Foreign Keys for X defined.  For The **API V1 the following OData queries work**:

$expand = Y
`http://localhost:59918/api/X?api-version=1.0&$expand=Y`

$expand = Z
`http://localhost:59918/api/X?api-version=1.0&$expand=Z`

$expand = Y,Z
`http://localhost:59918/api/X?api-version=1.0&$expand=Y%2CZ`

However, **only single expands work for V2 of the API**. I didn't change anything else in the way the API is configured between V1 & V2.

$expand = Y (WORKS)
`http://localhost:59918/api/X?api-version=2.0&$expand=Y`

$expand = Z (WORKS)
`http://localhost:59918/api/X?api-version=2.0&$expand=Z`

**$expand = Y,Z (DOESN'T WORK!)**
`http://localhost:59918/api/X?api-version=2.0&$expand=Y%2CZ`

The last query gives a "500 Undocumented | Error: Internal Server Error".

**Comment by commonsensesoftware** [comment]
What happens you try `http://localhost:59918/api/X?api-version=2.0&$expand=Y,CZ` directly in the browser? I feel like I've seen OData not be happy with the encoding before. If that works, you may need to tweak something in your OpenAPI/Swagger UI.

There should be a stack trace for the 500. Where is it failing? I'm willing to bet it's within the OData library.

**Comment by Clebeuf** [comment]
I just re-confirmed that changing the encoded , to an actual , in the query string didn't resolve the issue. If that was the issue, I would assume that V1 of the API would also have the same issue but that's working fine?

No exception is being thrown when debugging the API, and the api itself is simply returning "500 Undocumented | Error: Internal Server Error" with no stack trace.  I'll work on getting one.

**Comment by Clebeuf** [comment]
I am not able to get a stack trace for this issue since it's just returning a 500 with no additional information. Is there anything else that can provided to help with this investigation?

**Comment by commonsensesoftware** [comment]
Apologies for the long delay. Did you ever get this working?

I tried a couple more options and got it working with and without the encoding. Based on the information you've provided, I suspect that the issue is that the `$expand` option hasn't been enabled in this context. The matrix of global and scoped query options is super confusing and I've found it to sometimes be inconsistent.

By default, all query options are disabled (since OData 7.x I believe). I find it silly, but if I remember correctly, OData throws an exception which will propagate 500 if the client requests a query option that isn't supported. It _should_ be returning 400; the server didn't blow up, the client just asked for something that isn't allowed.  I'm willing to bet that if you simply enable the global query option, things will start working.

```c#
public void Configure( IApplicationBuilder app, VersionedODataModelBuilder modelBuilder )
{
    app.UseMvc(
        routeBuilder =>
        {
            // enable $expand globally
            routeBuilder.Expand();
            routeBuilder.MapVersionedODataRoutes( "odata", "api", modelBuilder.GetEdmModels() );
        } );
}
```

The API Explorer extension only reports which query options are allowed; it doesn't enforce anything. This means that it is possible to _describe_ a query option for an API that can't actually be invoked. The rules and processing are way more complicated than they should be, so I won't rule out a bug in the exploration process, but I suspect that isn't the case here.

Give that a whirl and let me know how it goes. Thanks.

**Comment by commonsensesoftware** [comment]
It seems this issue was abandoned or resolved. This _looks_ like an issue with the OData configuration. You can confirm this if a request is dispatched to your action. API Versioning doesn't get involved with OData policies outside of routing. This looks like something OData disallowed or is possibly a bug.

There were a number of issues related to OData API Exploration, but that would from discovering them, not executing them. Those should all be fixed in [5.0.0-RC.1](https://www.nuget.org/packages/Microsoft.AspNetCore.OData.Versioning/5.0.0-rc.1). The official release will likely occur after 2 weeks of _burn-in_. Feel free to reopen this if you continue to have an issue. Thanks.
